### PR TITLE
[GO-47] fix tests for race

### PR DESCRIPTION
**Pull request by jenchik**

### CODE REVIEW

**Comment by im-kulikov** [comment]
pls, fix 

https://github.com/cryptopay-dev/yaga/blob/7d7ca2ea10a7dc522ca27b58cf06160acf249b52/.circleci/config.yml#L48

must be
```
go test -race -count 1 ./..
```

**Review by im-kulikov — COMMENTED** [review]
pls, fix 

https://github.com/cryptopay-dev/yaga/blob/7d7ca2ea10a7dc522ca27b58cf06160acf249b52/.circleci/config.yml#L48

must be
```
go test -race -count 1 ./..
```

File: workers/waitgroup.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package workers
+
+import (
+	"runtime"
+
+	"go.uber.org/atomic"
+)
+
+type WaitGroup interface {
```
Comment (im-kulikov) [inline workers/waitgroup.go]: mb `Waiter`?

Comment (jenchik) [reply workers/waitgroup.go]: ok

Comment (jenchik) [reply workers/waitgroup.go]: but I think it's better to be ```WaitGroup``` as alternative sync.WaitGroup

**Review by im-kulikov — APPROVED** [review]

File: workers/waitgroup.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+func (wg *waitGroup) Add(delta int) {
+	wg.mu.Lock()
+	wg.count += int64(delta)
+	if wg.count == 0 {
+		wg.cond.Broadcast()
+	} else if wg.count < 0 {
+		wg.mu.Unlock()
+		panic("waitgroup: negative WaitGroup counter")
```
Comment (pcherednichenko) [inline workers/waitgroup.go:26]: are you sure that you should panic inside the package? this is bad practice

Comment (jenchik) [reply workers/waitgroup.go:26]: see https://github.com/golang/go/blob/master/src/sync/waitgroup.go#L74

**Review by pcherednichenko — APPROVED** [review]

**Pull request merged**